### ISSUE TITLE
cron schema not available in seed.sql (pg_cron)

### ISSUE THREAD

**Issue by zischler** [supabase/supabase#28966]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

When installing pg_cron via SQL (in the seed.sql) and afterwards scheduling a cron job (cron.schedule). I get the following error: `ERROR: schema "cron" does not exist`. When using the cron.schedule command later in the supabase sql editor it works.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:
1. Add following code to the seed.sql file of a clean supabase installation:
```
-- Check if pg_cron is available
DO $$
BEGIN
    -- Check if the pg_cron extension is available
    IF EXISTS (
        SELECT 1 FROM pg_available_extensions WHERE name = 'pg_cron'
    ) THEN
        -- Install pg_cron if it is available and not already installed (drop old one and install new)
        DROP EXTENSION IF EXISTS pg_cron CASCADE;
        CREATE EXTENSION pg_cron WITH SCHEMA pg_catalog; -- As seen in issue #28261

        -- Grant access to cron for user postgres
        GRANT USAGE ON SCHEMA cron TO postgres;
        GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA cron TO postgres;
    ELSE
        RAISE NOTICE 'pg_cron extension is not available in this PostgreSQL installation.';
    END IF;
END;
$$;

-- Verify pg_cron installation
DO $$
DECLARE
    installed_version TEXT;
BEGIN
    SELECT pae.installed_version INTO installed_version
    FROM pg_available_extensions AS pae
    WHERE pae.name = 'pg_cron';

    IF installed_version IS NOT NULL THEN
        RAISE NOTICE 'pg_cron extension installed with version: %', installed_version;
    ELSE
        RAISE NOTICE 'pg_cron extension is not installed.';
    END IF;
END;
$$;

-- STORED PROCEDURE (not really relevant)
CREATE PROCEDURE delete_old_profiles()
LANGUAGE plpgsql
AS $$
BEGIN
  -- First, delete users associated with profiles older than 30 days
  DELETE FROM auth.users AS u
  USING public."profiles" AS p
  WHERE p.user_id = u.id
  AND p.deleted_at < NOW() - INTERVAL '30 days';

  -- Then, delete the profiles that are older than 30 days
  DELETE FROM public."profiles" AS p
  WHERE p.deleted_at < NOW() - INTERVAL '30 days';
END;
$$;

-- create a cron job
SELECT cron.schedule('call-delete-old-profiles', '*/10 * * * *', 'CALL delete_old_profiles()');
```
2. See that pg_cron is correctly installed with version 1.6 (current version). 
3. But I still get following error:
```
ERROR: schema "cron" does not exist
```

## Expected behavior

I would expect supabase to have already correctly installed pg_cron and the postgres database to already have the schema cron ready to use. Using `SELECT cron.schedule('call-delete-old-profiles', '*/10 * * * *', 'CALL delete_old_profiles()');` would therefore schedule the job correctly.

## System information

- OS: macOS Sonoma
- Version of supabase-js: 2.45.2
- Version of Node.js: 20.10.0

## Additional context

Up until now I thought the best way to add cron jobs into our codebase would be through seed.sql or adding it to our Prisma migrations. We're using [Zenstack](https://github.com/zenstackhq/zenstack) for the project. Maybe there is another way, that is recommended to create cron jobs on server creation? This is currently pretty much missing in the docs.
If you think this issue has more to do with Postgres than Supabase, please let me know.

**Comment by encima** [comment]
Thanks for opening!

Note that you are installing `pg_cron` in the schema `pg_catalog`. Once installed, `pg_cron` will create the `cron` schema and it should be available. You can check this by running `\dn` using `psql` (or `select nspname from pg_catalog.pg_namespace;`)

If the `cron` schema is not created then it will fail or it may be that the search path is not including it for some reason.

Testing this locally, I am able to create the extension and jobs in the `seed.sql` (or subsequent migration files).

To debug, can you try running the same in a migration file and let us know the output please?

**Comment by zischler** [comment]
Thanks, I don't know why it didn't work on my machine. Now we got it to work by having the pg_cron installation in the seed.sql, and then after supabase seed is done we make another seed for all cronjobs.

We got it to work with using `create extension pg_cron with schema extensions` like in the current docs instead of `        CREATE EXTENSION pg_cron WITH SCHEMA pg_catalog;`, which was brought in issue #28261
```
/*
  ENABLE pg_cron extension
  https://supabase.com/docs/guides/database/extensions/pg_cron
*/
create extension pg_cron with schema extensions;
grant usage on schema cron to postgres;
grant all privileges on all tables in schema cron to postgres;
```

I'll close this issue, thanks for the quick reply. :)
